tests/test_insert.py

```diff
@@ -34,24 +36,34 @@ def test_insert_into_partitions():
         "INSERT INTO TABLE tab1 PARTITION (par1=1) SELECT * FROM tab2",
         {"tab2"},
         {"tab1"},
+        dialect="sparksql",
     )
 
 
 def test_insert_overwrite():
     assert_table_lineage_equal(
-        "INSERT OVERWRITE tab1 SELECT * FROM tab2", {"tab2"}, {"tab1"}
+        "INSERT OVERWRITE tab1 SELECT * FROM tab2",
+        {"tab2"},
+        {"tab1"},
+        dialect="sparksql",
     )
 
 
 def test_insert_overwrite_with_keyword_table():
     assert_table_lineage_equal(
-        "INSERT OVERWRITE TABLE tab1 SELECT col1 FROM tab2", {"tab2"}, {"tab1"}
+        "INSERT OVERWRITE TABLE tab1 SELECT col1 FROM tab2",
+        {"tab2"},
+        {"tab1"},
+        dialect="sparksql",
```

Comment:
I think the overall principle to test dialect specific feature is to list all dialect supported. In this case, we should test hive and sparksql.

Also maybe we should consider moving all the dialect specific test suits to a dedicated file.

Reply:
I agree with this. I have added some `_dialect_specific.py` tests for those tests where we are not testing ANSI queries.